### PR TITLE
Che ping/powerInfoPage

### PR DESCRIPTION
**Pull request by curry0622**
Complete power info page
* date picker
* mode button
* chart
* summary
* table
* styles
# Date Picker
* Let user be able to select a specific date in `powerInfoPage/datePicker.tsx`
# Mode Button
* Let user be able to change between `淨負載` and `產能設備` in `powerInfoPage/modeButton.tsx`
# Chart
* Draw a line chart based on fetched data
* Tooltip shows details of data
* All in `chart.tsx`
# Summary
* In `淨負載`, it shows a list of data in `powerInfoPage/summary/list.tsx`
* In `產能設備`, it shows a bar chart based on fetched data in `powerInfoPage/summary/barChart.tsx`
# Table
* Create a list based on fetched data
* User is able to switch pages
* All in `powerInfoPage/table.tsx`
# Styles
* Styles of power info page based on zeplin

### CODE REVIEW

File: src/components/powerInfoPage/datePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        : new Date(event.target.value);
+    changeDate(lastDate);
+  };
+
+  return (
+    <div className={classNames('powerinfo-datepicker-container')}>
+      <input
+        className={classNames('powerinfo-datepicker-input')}
+        type="date"
+        onChange={(event) => onChangeDate(event)}
```
Comment (coliver0221) [inline src/components/powerInfoPage/datePicker.tsx:26]: I think it's better to set the default value to show current date when user get into this page without choosing a date.
You can complete this by following these steps:
1. Create a variable to get current time from parent component
```tsx
// powerInfo/datePicker.tsx
interface IProps {
  changeDate(date: Date): void;
  currDate: Date;
}

const DatePicker: React.FC<IProps> = ({ changeDate, currDate }) => {
// ...
```
2. Pass the date in parent
```tsx
// powerInfo/index.tsx
<DatePicker changeDate={setDate} currDate={date} />
```
3. Use `dayjs` to format the variable and set to `value`
```tsx
// powerInfo/datePicker.tsx
import dayjs from 'dayjs';
// ...
<input
    className={classNames('powerinfo-datepicker-input')}
    type="date"
    value={dayjs(currDate).format('YYYY-MM-DD')}  // new
    onChange={(event) => onChangeDate(event)}
/>
```

File: src/components/powerInfoPage/chart.tsx [hunk 2]
```diff
@@ -0,0 +1,1784 @@
...
+  PV: number;
+  WT: number;
+}
+
+interface IProps {
+  mode: string;
+  date: Date;
+}
+
+interface IPadding {
```
Comment (coliver0221) [inline src/components/powerInfoPage/chart.tsx]: It seems like this `interface` didn't been used. Please delete it if it's useless.

File: src/components/powerInfoPage/table.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+  const clickLastPageHandler = () => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    setPage(lastPage);
+  };
+
+  const [prevDisabled, setPrevDisabled] = useState(true);
+  const [nextDisabled, setNextDisabled] = useState(false);
```
Comment (coliver0221) [inline src/components/powerInfoPage/table.tsx]: Please move these two line to the top of the functional component with other `useState` variables.

File: src/components/powerInfoPage/table.tsx [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+  useEffect(() => {
+    (async () => {
+      await fetchApiData();
+    })();
+    // eslint-disable-next-line @typescript-eslint/camelcase
+  }, [correctDate, per_page, page]);
+
+  const [timeText, setTimeText] = useState('紀錄時間');
+  const [typeText, setTypeText] = useState('用產電種類');
```
Comment (coliver0221) [inline src/components/powerInfoPage/table.tsx]: Please move these two line to the top of the functional component with other useState variables.

File: src/components/powerInfoPage/chart.tsx [hunk 5]
```diff
@@ -0,0 +1,1784 @@
...
+
+  // tick padding of axes
+  const [tickPaddingX, setTickPaddingX] = useState(0);
+  const [tickPaddingY, setTickPaddingY] = useState(0);
+
+  // bisector
+  const bisectDate = d3.bisector((d: IApiData) => d.Date).left;
+
+  // calculate first of the chart
+  const firstDate = new Date(lastDate.getTime() - 6 * 24 * 60 * 60 * 1000);
```
Comment (coliver0221) [inline src/components/powerInfoPage/chart.tsx]: It's better to use `dayjs.subtract()` to do operation on date.

File: src/pages/bidding.tsx [hunk 6]
```diff
@@ -4,7 +4,11 @@ import Layout from '../layouts/generalLayout';
 
 const BiddingPage: React.FC = () => {
   const { t } = useTranslation();
-  return <Layout title={t('biddingpage.pageTitle')}>BIDDING</Layout>;
+  return (
+    <Layout title={t('biddingpage.pageTitle')}>
+      <div />
+    </Layout>
+  );
```
Comment (coliver0221) [inline src/pages/bidding.tsx]: Why you modify this page in this PR?

File: src/components/powerInfoPage/chart.tsx (hunk 2 above)
Comment (curry0622) [reply src/components/powerInfoPage/chart.tsx]: Done. Thanks!

File: src/components/powerInfoPage/table.tsx (hunk 3 above)
Comment (curry0622) [reply src/components/powerInfoPage/table.tsx]: Done. Thanks!

File: src/components/powerInfoPage/table.tsx (hunk 4 above)
Comment (curry0622) [reply src/components/powerInfoPage/table.tsx]: Done. Thanks!

File: src/components/powerInfoPage/chart.tsx (hunk 5 above)
Comment (curry0622) [reply src/components/powerInfoPage/chart.tsx]: Done. Thanks!

File: src/pages/bidding.tsx (hunk 6 above)
Comment (curry0622) [reply src/pages/bidding.tsx]: I changed it back. Thanks!

File: src/components/powerInfoPage/datePicker.tsx (hunk 1 above)
Comment (curry0622) [reply src/components/powerInfoPage/datePicker.tsx:26]: Done. Thanks!

**Pull request merged**